.test-infra/jenkins/job_PreCommit_Java_Examples_Dataflow.groovy

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import PrecommitJobBuilder
+
+PrecommitJobBuilder builder = new PrecommitJobBuilder(
+    scope: this,
+    nameBase: 'Java Examples Dataflow',
+    gradleTask: ':beam-runners-google-cloud-dataflow-java-examples:preCommit',
```

Comment:
For other jobs we create a top-level `xyzPrecommit` task in the root build script. I don't know how the pattern was established, but it has some benefits:

1. Less logic in the Jenkins jobs, which aren't scoped to a commit and are more difficult to change. This way we can always refactor the project tree without messing with Jenkins.
1. Better discoverability, and easier to type than the super-long project name

Reply:
Ah, will do. Conversely, I think we should remove precommit/postcommit naming from everywhere else, instead giving the suites meaningful names. That way mechanism and policy are separate, and there's one place to look (the root).